src/components/shapes/draw.js

```diff
@@ -579,6 +599,224 @@ function setupDragElement(gd, shapePath, shapeOptions, index, shapeLayer, editHe
     }
 }
 
+function drawLabel(gd, index, options, shapeGroup) {
+    // Remove existing label
+    shapeGroup.selectAll('.shape-label').remove();
+
+    // If no label, return
+    if(!options.label) return;
```

Comment:
The `label` object is always present here; no?
We want to return when the `text` is empty.
```suggestion 
    if(!options.label.text) return;
```